### ISSUE TITLE
Issue: There are elements with invalid ARIA attributes or values. Ensure ARIA roles, states, and properties are valid.

### ISSUE THREAD

**Issue by A0707515** [primefaces/primeng#12950]
### Describe the bug

Hi @primeNG Team,

We need your support for the following accessibility violation found in Radio button element.

PrimeNg Library version : ^15.0.1

PrimeNg Radio button component : https://primeng.org/radiobutton

Live code example : https://codesandbox.io/s/dtdew7

---------------------------------------------------

**Issue :** There are elements with invalid ARIA attributes or values. 

**Examples include:** There are custom inputs with type="radio" and a redundant aria-checked attribute. native radio buttons do not require the aria-checked attribute.

**User Impact** : Assistive technologies may be unable to parse this content.

**[Code Reference]**
<input type="radio" id="ingredient1" name="pizza" value="Cheese" **aria-checked="true"**>

Please refer below given screenshot:

![image](https://user-images.githubusercontent.com/121850445/234516996-2acb347e-95b1-4133-ac42-95bd457d8af6.png)

**[Recommendation]**

Ensure ARIA roles, states, and properties are valid. Many ARIA attributes can only be used on certain elements. Some can only be used in combination with other attributes. Additionally, if an attribute references an ID, that ID must be present in the page. To detect many of these errors automatically, use the Nu Html Checker: https://validator.w3.org/nu/

**[Compliant Code Example]**

<input type="radio" id="ingredient1" name="pizza" value="Cheese">

Can you please evaluate it and provide the generic solution on this.

Please do let me know if additional details are required to be added.

Thanks,
Jignesh Raval

### Environment

Angular 15
Node 16
Windows 10

### Reproducer

https://codesandbox.io/s/dtdew7

### Angular version

"@angular/animations": "^15.0.2",     "@angular/cdk": "^15.0.1",     "@angular/common": "^15.0.2",     "@angular/compiler": "^15.0.2",     "@angular/core": "^15.0.2",     "@angular/forms": "^15.0.2",     "@angular/platform-browser": "^15.0.2",     "@angular/platform-browser-dynamic": "^15.0.2",     "@angular/router": "^15.0.2",     "@angular/service-worker": "^15.0.2",

### PrimeNG version

15.0.1

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

16

### Expected behavior

If the element is self explanatory then there is no need to add additional aria roles and attributes like Radio button, checkbox etc...

**Note:** As per https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/radio_role - The first rule of ARIA is if a native HTML element or attribute has the semantics and behavior you require, use it instead of re-purposing an element and adding ARIA. Instead use the native [HTML <input type="radio">](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio) (with an associated [<label>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label), which natively provides all the functionality required:

Ensure ARIA roles, states, and properties are valid. Many ARIA attributes can only be used on certain elements. Some can only be used in combination with other attributes. Additionally, if an attribute references an ID, that ID must be present in the page. To detect many of these errors automatically, use the Nu Html Checker: https://validator.w3.org/nu/

**Comment by mertsincan** [comment]
Hi,

So sorry for the delayed response! Improvements have been made to many components recently, both in terms of performance and enhancement. Therefore, this improvement may have been developed in another issue ticket without realizing it. You can check this in the documentation. If there is no improvement on this, can you open a new issue so we can include it in our roadmap?

Thanks a lot for your understanding!
Best Regards,
